sdk/core/core-amqp/review/core-amqp.api.md

```diff
@@ -61,9 +61,9 @@ export interface CbsResponse {
     // (undocumented)
     correlationId: string;
     // (undocumented)
-    satusDescription: string;
-    // (undocumented)
     statusCode: string;
+    // (undocumented)
+    statusDescription: string;
```

Comment:
@richardpark-msft I wonder if this could break anything else.

Reply:
Yes, this would be a breaking change. If we are doing this, then we will have to update the major version to 2

Reply:
Gah, the curse of having an "internal" library be externally available.

Everything else in this PR is goodness, let's just leave this change out for now.

Reply:
I don't know...I kind of see it as a bug fix. I think the fact that no one has reported that it's typo'd as `satusDescription` is some indicator that probably no one is using this field yet. If we really wanted, we could create an alias in JS so anyone using that field at runtime would still see it. Though having done something like that in the past people will wonder why they both show up...

Reply:
I think we should fix this - it's not super clear that this is typo'd, I might just not even realize what it is.

There is a hybrid approach here - we can support both, but have the typings (and docs) only show latest. That way JS users don't need to do anything, and TS builds break but the fix is easy - it will offer a "did you mean" error since the string is only off-by-one, and the user could just cast to any without any code changes to make the error go away and things work...

Reply:
@bterlson Would you be opposed to making the old field non-enumerable in JS as well? That way if you are accessing it directly you'll continue to be able to do so, but you won't see it if you just print the object or enumerate over the fields.

Reply:
A new proposal: TypeScript will soon learn to display members with @deprecated tag in a way that makes it obvious the thing is deprecated (at least in vscode). We can do a more conservative thing in that case:

1. Add statusDescription
2. Deprecate satusDescription by adding @​deprecated. Ensure docs will call out deprecated members/hide by default. Install it as a non-enumerable getter that logs a warning on access.
3. Remove satusDescription in next major.

This will have the unfortunate effect of showing the deprecated member until likely TS4.0, but doesn't break anyone and makes sure the good name is present for those that don't want to opt in to a future break by using a typo'd property and likely reduces confusion overall.

Given that this is core-amqp and it's unlikely anyone but us is using it directly, this can probably wait for next major too.

Reply:
Wow! Thank you for this. This is a much clearer direction.

Reply:
@bterlson how do I create a non-enumerable getter? I can think of using a decorator, or adding the getter to the prototype of the type, both seem weird.

Reply:
Also, what do you all think of this message? `CbsResponse's property "satusDescription" is deprecated, use "statusDescription" instead.`

Reply:
If today you have:

```js
let obj = {
  ...
  satusDescription: str
}
```

you would change it to
```js
let obj = {
  statusDescription: str
}

Object.defineProperty(obj, 'satusDescription', {
  get() {
    logger.warning(`CbsResponse's property "satusDescription" has been renamed to "statusDescription". "satusDescription" will be removed in v2.0.0.`)
    return this.statusDescription;
  }
});
```

Reply:
While taking this approach makes sense for when we want to deprecate properties, there is not much of a return on investment here. There most likely is no user scenario to even expose the statusCode and the statusDescription in this case.

For the scope of this PR, I would recommend not making any change to `satusDescription`. @richardpark-msft, I would recommend we log an issue to investigate the usefulness of the status code & description when getting back the response of a claim negotiation. I am sure that a deeper dive into core-amqp may bring other things we want to change in a v2

Reply:
Alright, I'm undoing that specific change then.

@bterlson Thank you for that though! I haven't used `defineProperty` in ages 😱 amazing.

Reply:
@richardpark-msft Here's an issue! https://github.com/Azure/azure-sdk-for-js/issues/9031 Good luck 🌞